realm/realm-library/src/androidTest/java/io/realm/RealmIntegerTests.java

```diff
@@ -0,0 +1,216 @@
+                new Realm.Transaction() {
+                    @Override
+                    public void execute(Realm realm) {
+                        ri.decrement(1);
+                    }
+                });
+
+        allTypes = realm.where(AllTypes.class).findFirst();
+        assertEquals(allTypes.getColumnRealmInteger().longValue(), 4);
+    }
+
+
+    @Ignore("not yet implemented")
+    @Test
+    public void testCopyTo() {
+        final AllTypes allTypes = new AllTypes();
+        RealmInteger ri = new RealmInteger(5);
+        allTypes.setColumnRealmInteger(ri);
+
+        realm.executeTransaction(
+                new Realm.Transaction() {
+                    @Override
+                    public void execute(Realm realm) {
+                        realm.copyToRealm(allTypes);
+                    }
+                });
+
+        realm.executeTransaction(
+                new Realm.Transaction() {
+                    @Override
+                    public void execute(Realm realm) {
+                        realm.where(AllTypes.class).findFirst().getColumnRealmInteger().decrement(1);
+                    }
+                });
+
+        assertEquals(ri.longValue(), 4);
+    }
+
+    @Ignore("not yet implemented")
+    @Test
+    public void testCopyFrom() {
+        realm.executeTransaction(
+                new Realm.Transaction() {
+                    @Override
+                    public void execute(Realm realm) {
+                        realm.createObject(AllTypes.class).setColumnRealmInteger(new RealmInteger(5));
+                    }
+                });
+
+        realm.beginTransaction();
+        AllTypes unmanagedAllTypes = realm.copyFromRealm(realm.where(AllTypes.class).findFirst());
+        realm.commitTransaction();
+
+        realm.executeTransaction(
+                new Realm.Transaction() {
+                    @Override
+                    public void execute(Realm realm) {
+                        realm.where(AllTypes.class).findFirst().getColumnRealmInteger().decrement(1);
+                    }
+                });
+
+        assertEquals(unmanagedAllTypes.getColumnRealmInteger().longValue(), 5);
+    }
+
+    @Ignore("not yet implemented")
+    @Test
+    public void testUpdateManagedOutsideTransactionFails() {
+        realm.executeTransaction(
+                new Realm.Transaction() {
+                    @Override
+                    public void execute(Realm realm) {
+                        realm.createObject(AllTypes.class).setColumnRealmInteger(new RealmInteger(5));
+                    }
+                });
+
+        RealmInteger ri = realm.where(AllTypes.class).findFirst().getColumnRealmInteger();
+        try {
+            ri.decrement(1);
+            fail("Mutating a managed RealmInteger outside a transaction should fail");
+        } catch (IllegalStateException e) {
+            assertTrue(e.getMessage().contains("must be in a transaction"));
+        }
+    }
+}
```

Comment:
To be fair. `compareTo` has a complex enough contract to warrant unit tests. The aim is 100% coverage, so not doing unit tests just because "the implementation is simple" is not an argument. It might change down the line.